src/main/java/frc/robot/subsystems/claw/ClawSubsystem.java

```diff
@@ -0,0 +1,75 @@
+package frc.robot.subsystems.claw;
+
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import edu.wpi.first.wpilibj.Solenoid;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+
+import static frc.robot.Constants.ClawConstants.*;
+
+public class ClawSubsystem extends SubsystemBase {
+  private TalonSRX rightMotor;
+  private TalonSRX leftMotor;
+  private Solenoid pfft;
+
+  // Set initial state: claw open and not lifted
+  public boolean clawIsOpen = true;
+  public boolean clawIsLifted = false;
+
+  public ClawSubsystem() {
+    CommandScheduler.getInstance().registerSubsystem(this);
+
+    // Initialize motors and pneumatic
+    rightMotor = new TalonSRX(rightMotorPort);
+    leftMotor = new TalonSRX(leftMotorPort);
+
+    pfft = new Solenoid(pfftPCMPort);
+
+    // Configure the motors for the encoder
+    rightMotor.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
+    rightMotor.setSelectedSensorPosition(0);
+  }
+
+  @Override
+  public void periodic() {
+    // If claw is meant to be open
+    if (clawIsOpen) {
+
+      // Set motor
+      if (rightMotor.getSelectedSensorPosition() < clawOpenPosition) {
+        rightMotor.set(ControlMode.PercentOutput, clawMotorSpeed);
```

Comment:
this might work for now. for future projects, we should use PIDs to get to a position. talon also has a built-in runmode that tries to get to a position, I think.

But in my opinion if we ever had to write competition code for a mech like this, the most applicable solution might be to set power when closing until the motor stalls out (it has a 0 rate of change in position)

Reply:
created isMotorStalled() function that checks if the change in motor position is below a certain delta value; this method is called to check when to stop closing the claw.